### ISSUE TITLE
xhrStream does not use options from IProvider.preCall

### ISSUE THREAD

**Issue by mictrax** [ewsjs/xhr#14]
In principle, it should be possible to change the request options (`IXHROptions`) with `IProvider.preCall`. This works perfectly for `xhr` but not for `xhrStream`.  When given an authProvider, the preCall function is called and returns an options promise. But these options are not taken into account on the request:

```
xhrStream(xhroptions: IXHROptions, progressDelegate: (progressData: IXHRProgress) => void): Promise<XMLHttpRequest> {
...
  options = this.getOptions(options)
  return new Promise<XMLHttpRequest>((resolve, reject) => {
    let _promise: Promise<IXHROptions> = Promise.resolve(options);

    if (this.authProvider) {
      _promise = this.authProvider.preCall(options);
    }

    _promise.then(opt => {
      this.stream = request(options);
...
``` 

Imho, the options derived from the preCall function should be used in `xhrStream` in the same way as for `xhr`.

E.g.:
```
xhrStream(xhroptions: IXHROptions, progressDelegate: (progressData: IXHRProgress) => void): Promise<XMLHttpRequest> {
...
  options = this.getOptions(options)
  return new Promise<XMLHttpRequest>((resolve, reject) => {
    let _promise: Promise<IXHROptions> = Promise.resolve(options);

    if (this.authProvider) {
      _promise = this.authProvider.preCall(options);
    }

    _promise.then(opt => {
      this.stream = request(opt || options);
...
```

**Comment by cm253** [comment]
@gautamsi Can we also have a bugfix release once the PR has been merged? We are using a workaround, but would be better to have this fixed in the library.

**Comment by gautamsi** [comment]
I will create release tomorrow

**Comment by mictrax** [comment]
@gautamsi Thanks for merging the PR. Will there also be a bugfix release soon?

**Comment by gautamsi** [comment]
have published 1.5.0 now, please provide feedback on this. typescript compiling has done greater changes than the actual PR, which should work fine.

**Comment by mictrax** [comment]
Thanks a lot. The release works fine. Workaround could be removed.
